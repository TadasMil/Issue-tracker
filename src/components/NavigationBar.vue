<script setup lang="ts" name="NavigationBar">
import type { IPage } from '@/types'
import { useRoute } from 'vue-router'

const pages: IPage[] = [
  { name: 'Home', path: '/' },
  { name: 'Open Issues', path: '/issues/open' },
  { name: 'Completed Issues', path: '/issues/completed' },
  { name: 'Trashed Issues', path: '/issues/trashed' },
]

const route = useRoute()

const isActive = (page: IPage) => route.path === page.path
</script>

<template>
  <b-nav class="bg-dark">
    <b-nav-item v-for="page in pages" :key="page.path" :to="page.path" :active="isActive(page)">
      <b-nav-text class="text-light">{{ page.name }}</b-nav-text>
    </b-nav-item>
  </b-nav>
</template>

<style #scoped>
.active {
  background-color: var(--bs-secondary);
}
</style>
