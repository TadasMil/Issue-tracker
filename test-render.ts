import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
