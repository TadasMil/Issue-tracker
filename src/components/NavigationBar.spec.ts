import { describe, it, expect, vi } from 'vitest'
import { render } from '../../test-render'

import NavigationBar from './NavigationBar.vue'

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => ({ path: '/' })),
}))

describe('NavigationBar', () => {
  it('should render the component given home page is active', () => {
    const { html } = render(NavigationBar)

    expect(html()).toMatchSnapshot()
  })
})
