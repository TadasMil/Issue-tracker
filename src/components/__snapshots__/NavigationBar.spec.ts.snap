// Vitest Snapshot v1

exports[`NavigationBar > should render the component given home page is active 1`] = `
"<ul class=\\"nav bg-dark\\">
  <li class=\\"nav-item\\"><a class=\\"active nav-link\\" to=\\"/\\" href=\\"/\\" target=\\"_self\\">
  <li class=\\"navbar-text text-light\\">Home</li></a></li>
  <li class=\\"nav-item\\"><a class=\\"nav-link\\" to=\\"/issues/open\\" href=\\"/issues/open\\" target=\\"_self\\">
  <li class=\\"navbar-text text-light\\">Open Issues</li></a></li>
  <li class=\\"nav-item\\"><a class=\\"nav-link\\" to=\\"/issues/completed\\" href=\\"/issues/completed\\" target=\\"_self\\">
  <li class=\\"navbar-text text-light\\">Completed Issues</li></a></li>
  <li class=\\"nav-item\\"><a class=\\"nav-link\\" to=\\"/issues/trashed\\" href=\\"/issues/trashed\\" target=\\"_self\\">
  <li class=\\"navbar-text text-light\\">Trashed Issues</li></a></li>
</ul>"
`;
